File: apps/api/app/schemas/feedback.py
from typing import Literal
from app.schemas.pagination import PaginatedResponse
from pydantic import BaseModel

FeedbackSentiment = Literal["Positive", "Neutral", "Negative"]
CustomerTier = Literal["Enterprise", "Growth", "Startup"]
FeedbackUrgency = Literal["High", "Medium", "Low"]


class FeedbackCreate(BaseModel):
    customer: str
    request: str
    product_area: str
    sentiment: FeedbackSentiment
    tier: CustomerTier
    urgency: FeedbackUrgency
    source: str
    linked_feature: str
    received_at: str


class FeedbackUpdate(BaseModel):
    customer: str | None = None
    request: str | None = None
    product_area: str | None = None
    sentiment: FeedbackSentiment | None = None
    tier: CustomerTier | None = None
    urgency: FeedbackUrgency | None = None
    source: str | None = None
    linked_feature: str | None = None
    received_at: str | None = None


class FeedbackItem(BaseModel):
    id: str
    customer: str
    request: str
    product_area: str
    sentiment: FeedbackSentiment
    tier: CustomerTier
    urgency: FeedbackUrgency
    source: str
    linked_feature: str
    received_at: str
    # Allow converting SQLAlchemy model objects into API response objects.
    model_config = {"from_attributes": True}


FeedbackListResponse = PaginatedResponse[FeedbackItem]

File: apps/web/src/types/feedback.ts
import type { PaginatedResponse, PaginationParams } from "./pagination";

type FeedbackSentiment = "Positive" | "Neutral" | "Negative";
type FeedbackUrgency = "High" | "Medium" | "Low";
type CustomerTier = "Enterprise" | "Growth" | "Startup";

type FeedbackInboxItem = {
  id: string;
  customer: string;
  request: string;
  productArea: string;
  sentiment: FeedbackSentiment;
  tier: CustomerTier;
  urgency: FeedbackUrgency;
  source: string;
  linkedFeature: string;
  receivedAt: string;
};

type FeedbackCreateInput = Omit<FeedbackInboxItem, "id">;

type FeedbackUpdateInput = Partial<FeedbackCreateInput>;

type FeedbackListFilters = {
  productArea?: string;
  search: string;
  urgency?: FeedbackUrgency;
};

// Use generic pagination type
type FeedbackListResponse = PaginatedResponse<FeedbackInboxItem>;

type FeedbackSummary = {
  label: string;
  value: string;
  helper: string;
  icon: string;
};

export type {
  CustomerTier,
  FeedbackCreateInput,
  FeedbackInboxItem,
  FeedbackListFilters,
  FeedbackListResponse,
  FeedbackSentiment,
  FeedbackSummary,
  FeedbackUrgency,
  FeedbackUpdateInput,
  PaginationParams,
};
